src/OpenTelemetry/ApiCompatBaseline.txt

```diff
@@ -0,0 +1,4 @@
+Compat issues with assembly OpenTelemetry:
+CannotChangeAttribute : Attribute 'System.Runtime.CompilerServices.NullableAttribute' on generic param 'TInstrumentation' on member 'OpenTelemetry.Metrics.MeterProviderBuilderBase.AddInstrumentation<TInstrumentation>(System.Func<TInstrumentation>)' changed from '[NullableAttribute((byte)0)]' in the contract to '[NullableAttribute((byte)2)]' in the implementation.
+CannotChangeAttribute : Attribute 'System.Runtime.CompilerServices.NullableAttribute' on generic param 'TInstrumentation' on member 'OpenTelemetry.Trace.TracerProviderBuilderBase.AddInstrumentation<TInstrumentation>(System.Func<TInstrumentation>)' changed from '[NullableAttribute((byte)0)]' in the contract to '[NullableAttribute((byte)2)]' in the implementation.
+Total Issues: 2
```

Comment:
This was very tricky. I had to get help from @ericstj.

Short version: ApiCompat detects that something has changed but it doesn't attempt to figure out if the change is safe.

Long version:

What is going on here is explained by [Nullable Metadata](https://github.com/dotnet/roslyn/blob/main/docs/features/nullable-metadata.md#type-parameters).

Our API before was...

```csharp
    public abstract TracerProviderBuilder AddInstrumentation<TInstrumentation>(
        Func<TInstrumentation> instrumentationFactory)
        where TInstrumentation : class;
```

That is WITHOUT a nullable context so `TInstrumentation` in `TracerProviderBuilderBase` got `[NullableAttribute((byte)0)]` in itsmetadata.

Now we have annotations & a nullable context for this API as:

```csharp
    public abstract TracerProviderBuilder AddInstrumentation<TInstrumentation>(
        Func<TInstrumentation> instrumentationFactory)
        where TInstrumentation : class?;
```

Therefore `TInstrumentation` in `TracerProviderBuilderBase` gets changed to `[NullableAttribute((byte)2)]`.

Both are null-allowing just one is "without annotations" and one is "with annotations."

My understanding is this is safe.

Reply:
Thanks @ericstj!